### ISSUE TITLE
not support comment generation for functions that return pointer

### ISSUE THREAD

**Issue by mano233** [kkoomen/vim-doge#179]
**Describe the bug**
The comment of the function that returns the pointer does not generate "@param"

**Settings**
use default setting

**To Reproduce**

- Create file `demo.cpp` with:

```cpp
int* fun1(int a,int b,int c){
    return nullptr;
}

int fun2(int a,int b,int c){
    return 0;
}

```

- Start (neo)vim with command: `<leader>d`

**Expected behavior**
the fun1 should generate "@param",but.....
![image](https://user-images.githubusercontent.com/29190675/100957774-13c3f100-3556-11eb-8928-26bdac5c80dc.png)

**Output of `vim --version`**

```shell
NVIM v0.4.4
Build type: Release
LuaJIT 2.0.5
Compilation: /usr/bin/clang -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DNDEBUG -DMIN_LOG_LEVEL=3 -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -I/tmp/neovim-20201114-45415-1dcr9av/neovim-0.4.4/build/config -I/tmp/neovim-20201114-45415-1dcr9av/neovim-0.4.4/src -I/usr/local/include -I/tmp/neovim-20201114-45415-1dcr9av/neovim-0.4.4/deps-build/include -I/Library/Developer/CommandLineTools/SDKs/MacOSX11.0.sdk/usr/include -I/usr/local/opt/gettext/include -I/tmp/neovim-20201114-45415-1dcr9av/neovim-0.4.4/build/src/nvim/auto -I/tmp/neovim-20201114-45415-1dcr9av/neovim-0.4.4/build/include
Compiled by brew@BigSur

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/local/Cellar/neovim/0.4.4/share/nvim"

Run :checkhealth for more info
```

**Comment by kkoomen** [comment]
Indeed a bug, because I never thought about this scenario personally. I will fix this issue today somewhere or otherwise this weekend definitely.

**Comment by kkoomen** [comment]
This bug has been fixed and released in [v3.6.0](https://github.com/kkoomen/vim-doge/tree/v3.6.0).

Feel free to submit any new issues if you experience any unwanted behavior in the future. Thanks for your contribution.
